src/animation/gravity.ts

```diff
@@ -36,31 +49,62 @@ const applyGravitationalForceForStep = (
   });
 };
 
-export interface Gravity1DParams extends AnimationParams {
-  config: {
-    moverMass: number;
-    attractorMass: number;
-    r: number;
-  };
-  onUpdate: VectorSetter;
-  onComplete: () => void;
-}
+const reversePlayState = (
+  state: Gravity1DState,
+  config: Gravity1DParams['config']
+) => {
+  if (state.mover.position[0] > config.r) {
+    state.mover = {
+      ...state.mover,
+      acceleration: [0, 0],
+      velocity: [0, 0],
+      position: [config.r, 0],
+    };
+    state.attractor = {
+      ...state.attractor,
+      position: [0, 0],
+    };
```

Comment:
In the case of gravity, when the animation completes we essentially transport the `attractor` to [0, 0] so the `mover` will reverse course and get pulled towards it.